frontend/src/metabase/containers/UndoListing.jsx

```diff
@@ -126,4 +159,43 @@ function UndoListingInner() {
   );
 }
 
+function UndoProgress({ paused, timeout: initialTimeout }) {
+  const [value, setValue] = useState(100);
+  // timeout of undo will change after pause/resume, but it shouldn't affect
+  // progress bar step
+  const [timeout] = useState(initialTimeout);
+  const [isRunning, setIsRunning] = useState(true);
+  const transitionTime = 100; // default value in Progress in v7
+  const step = 100 / (timeout / transitionTime);
+
+  useInterval(
```

Comment:
> I still can see toast to disappear a bit earlier than a bar, but now no jumps

Unfortunately it's more noticeable than before. I believe this `useInterval` + `step` is the reason.

Did you consider using `useRaf` instead? That should iron out the sync differences between intervals.

We should be able to compute and remember what the last known progress is on `PAUSE_UNDO` action, and then this component could be simplified to something like this:

```tsx
const elapsed = useRaf(timeout);
const progressRemaining = 100 - lastProgress;
const value = paused ? progressRemaining : progressRemaining * (1 - elapsed);
```

WDYT?

note: we'd need to disable the `transition` on `[role="progressbar"]` element - otherwise updates would be too fast for the CSS animation to even begin.

Reply:
https://github.com/metabase/metabase/pull/44015/commits/ce8f370f3d79ab6cbfa94a67bc21d1ff2dc15225